detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/ViewCommandOperationsReflected.java

```diff
@@ -0,0 +1,65 @@
+
+import org.joor.Reflect;
+
+class ViewCommandOperationsReflected {
+    private static final String LOG_TAG = "Detox";
+    private final static String CLASS_DISPATCH_COMMAND_OPERATION =
+            "com.facebook.react.uimanager.UIViewOperationQueue$DispatchStringCommandOperation";
+    private final static String FIELD_MVIEW_COMMAND_OPERATIONS = "mViewCommandOperations";
+    private final static String FIELD_NUM_RETRIES = "numRetries";
+    private final static String FIELD_MTAG = "mTag";
+    private final static String FIELD_MCOMMAND = "mCommand";
+    private final static String FIELD_MVIEW_MANAGERS = "mNativeViewHierarchyManager";
+    private final static String SET_NATIVE_VALUE = "setNativeValue";
+    private final static String REACT_SWITCH = "ReactSwitch";
+    private final static int NUM_TIMES_BEFORE_NOTIFY_IDLE = 10;
+    private int timesStuckQueueDetected = 0;
+
+    // This is a workaround for https://github.com/facebook/react-native/issues/32594
+    // uses duck typing heuristics to determine that this is probably the stuck Switch operation and if so, ignores it
+    boolean workaroundForRN66Bug(Object uiOperationQueue) {
+        boolean isStuckSwitchOperation = false;
+
+        try {
+            Class<?> DispatchStringCommandOperation = Class.forName(CLASS_DISPATCH_COMMAND_OPERATION);
+            List<?> mViewCommandOperations = Reflect.on(uiOperationQueue).field(FIELD_MVIEW_COMMAND_OPERATIONS).get();
+
+            if (mViewCommandOperations.size() == 1) {
+                Object viewOperation = DispatchStringCommandOperation.cast(mViewCommandOperations.get(0));
+
+                int mTag = Reflect.on(viewOperation).field(FIELD_MTAG).get();
+                UIViewOperationQueue uiManagerModule = (UIViewOperationQueue) uiOperationQueue;
+                NativeViewHierarchyManager nativeHierarchyManager = Reflect.on(uiManagerModule).field(FIELD_MVIEW_MANAGERS).get();
+                String mClassName = nativeHierarchyManager.resolveView(mTag).getClass().getSimpleName();
+
+                int numRetries = Reflect.on(viewOperation).field(FIELD_NUM_RETRIES).get();
+                String mCommand = Reflect.on(viewOperation).field(FIELD_MCOMMAND).get();
+
+                boolean isReactSwitch = mClassName.equals(REACT_SWITCH);
+                boolean hasOneRetryIncremented = numRetries == 1;
+                boolean isSetNativeValueCommand = mCommand.equals(SET_NATIVE_VALUE);
+
+                if (isReactSwitch && hasOneRetryIncremented && isSetNativeValueCommand) {
+                    timesStuckQueueDetected++;
+                }
+
+                if (timesStuckQueueDetected >= NUM_TIMES_BEFORE_NOTIFY_IDLE) {
+                    isStuckSwitchOperation = true;
+                }
+            } else {
+                timesStuckQueueDetected = 0;
+            }
+        } catch (ClassNotFoundException e) {
+            Log.d(LOG_TAG, String.valueOf(e.getCause()));
+        }
+        return isStuckSwitchOperation;
```

Comment:
This is a step in the right direction, but it still requires more work.

1. The point behind `XXXReflected` classes is to provide an easier access into the inner fields using reflection (i.e. abstract away the ugly reflection details). This concept is still not utilized well enough in this case. We need something more in the likes of `UIManagerModuleReflected`, `NativeHierarchyManagerReflected` and `ViewOperationReflected` classes, for that. Once applied, we must have the reflection details and the workaround logic *separated* into different classes / methods.
1. I think that detecting whether _an_ "anonymous" suspicious command is repeatedly stuck in the queue is okay, but it can and should be made more robust: We should try harder and see whether _the_ suspicious command is stuck, by repeatedly comparing whether the current command is identical to our temp (weak) reference to it.